.github/workflows/meson_ci.yml

```diff
@@ -0,0 +1,124 @@
+## This Workflow just will run when a PR has "meson" label.
+
+name: Dinit Meson based CI
+
+on:
+  pull_request:
+    branches:
+      - master
+    paths:
+      - '**'
+      - 'build/version.conf'
+      - '!.*'
+      - '!README.md'
+      - '!NEWS'
+      - '!BUILD*'
+      - '!CONTRIBUTORS'
+      - '!TODO'
+      - '!LICENSE'
+      - '!**/Makefile'
+      - '!build/**'
+      - '!doc/CODE-STYLE'
+      - '!doc/COMPARISON'
+      - '!doc/CONTRIBUTING'
+      - '!doc/DESIGN'
+      - '!doc/getting_started.md'
+      - '!doc/manpages/README'
+      - '!doc/manpages/generate-html.sh'
+      - '!doc/linux/**'
+  workflow_dispatch:
+
+jobs:
+
+  Debian-bullseye_build:
+
+    if: ${{ github.run_attempt >= 2 || contains(github.event.pull_request.labels.*.name, 'meson') }}
```

Comment:
So *any* rerun of CI will always run the meson CI builds? Is that what you intended?

Reply:
Could we just leave the condition as it was ("pull request label contains 'meson'") and then if we want to run meson CI for a given pull request just add the label and then re-run CI?

Reply:
> So _any_ rerun of CI will always run the meson CI builds? Is that what you intended?

Any rerun of **Meson** CI will always run the meson CI builds.

Reply:
> Could we just leave the condition as it was ("pull request label contains 'meson'") and then if we want to run meson CI for a given pull request just add the label and then re-run CI?

I tested it before but it's need an another commit to work.

Reply:
> Any rerun of Meson CI will always run the meson CI builds.

But the condition is "or":

```
    if: ${{ github.run_attempt >= 2 || contains(github.event.pull_request.labels.*.name, 'meson') }}
```

If this is the 2nd run, OR the pull event contains 'meson'

Reply:
**Regular CI** and **Meson based CI** are two different workflows and rerun of one of them, doesn't rerun other workflow.

Reply:
Alright, so how can you re-run the meson workflow if it never ran the first time?

Reply:
1. a PR is related to meson but it doesn't have meson label.
2. Meson CI skipped for first time.
3. one of the maintainers rerun skipped workflow. it's will make `github.run_attempt` = 2.
4. Meson CI will run for that PR. Also we add meson label for auto meson ci.

Reply:
Why not:

1. a PR is related to meson but it doesn't have meson label.
2. Meson CI skipped for first time.
3. one of the maintainers adds the  meson label
4. maintainer then reruns the skipped workflow

Doesn't that achieve the same thing?

Reply:
5. Unfortunately it's skipped again due to some reasons.

Reply:
> > So _any_ rerun of CI will always run the meson CI builds? Is that what you intended?
> 
> Any rerun of **Meson** CI will always run the meson CI builds.

I don't see how to run just Meson CI. Github UI says "rerun all jobs". It's not specific to a workflow.

![image](https://github.com/davmac314/dinit/assets/8269006/2727d39f-bb4f-4d14-b20f-5825fa247f68)

Reply:
So if the condition is as it is currently, re-running jobs will run the meson CI workflow even if it isn't a meson-related pull request.

Reply:
Hmm, maybe I'm wrong. Maybe it separates different workflows into different jobs. But I want to check more, will need to do that later.

Reply:
For sure, I reran Regular CI for your last commit. As you see It doesn't trigger Meson based CI.
https://github.com/davmac314/dinit/actions/runs/5115941484

Reply:
Ok, sounds good.

Can you please put a comment in the file explaining all this (just a couple of sentences is fine).

Can we make it run on any change to master (once merged/pushed)? I think it's good to know if meson build does break

Reply:
Details: https://blog.mergify.com/running-github-actions-only-on-certain-pull-requests/

Reply:
I think we need to bypass that check in workflow for pushes, No? Currently it needs a re-run request to run.

Reply:
> Currently it needs a re-run request to run.

I think with the correct dispatch (`on: pull_request: types: - labeled') it won't need a re-run request.

But yes, the conditions may need to be different for a pull request vs push to master.

Reply:
But seems like it doesn't work on newer commits!

Reply:
you need it dispatched on the default event types too - `opened`, `reopened` and `synchronize`. If the event type is `label` it only triggers at the time the label is added.